src/slice.rs

```diff
@@ -157,7 +157,8 @@ impl<'a, T> NonEmptyIterator for NEChunks<'a, T> {
         } else if self.index >= self.tail.len() {
             None
         } else {
-            let end = self.index + self.window.get();
+            // Ensure we never go out of bounds
+            let end = min(self.index + self.window.get(), self.tail.len());
             let slc: &'a [T] = &self.tail[self.index..end];
```

Comment:
My incorrect assumption was based on this:
```rust
fn main() {
    let v = vec![1, 2, 3];
    let u = &v[0..];

    println!("{:?}", u);
}
```
This does not panic. But this does:
```rust
fn main() {
    let v = vec![1, 2, 3];
    let u = &v[0..10];

    println!("{:?}", u);
}
```